### PR TITLE
feat: support opentracing via jaeger-client-go

### PR DESCRIPTION
**Pull request by hairmare**

### CODE REVIEW

**Comment by eyenx** [comment]
my :eyes: on it

File: pkg/mongodb/tracer/tracer.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+func (t *tracer) HandleFailedEvent(ctx context.Context, evt *event.CommandFailedEvent) {
+	if evt == nil {
+		return
+	}
+	if rawSpan, ok := t.spans.Load(evt.RequestID); ok {
+		defer t.spans.Delete(evt.RequestID)
+		if span, ok := rawSpan.(opentracing.Span); ok {
+			defer span.Finish()
+			ext.Error.Set(span, true)
+			span.LogFields(log.String("failure", evt.Failure))
```
Comment (eyenx) [inline pkg/mongodb/tracer/tracer.go:67]: No duration tag when failed?

Comment (hairmare) [inline pkg/mongodb/tracer/tracer.go:67]: like so?
```suggestion
			span.SetTag(prefix+"duration", evt.DurationNanos)
			span.LogFields(log.String("failure", evt.Failure))
```

Comment (eyenx) [reply pkg/mongodb/tracer/tracer.go:67]: yeah why not? or is this dumb?

Comment (hairmare) [reply pkg/mongodb/tracer/tracer.go:67]: took it for a spin before adding it

**Review by eyenx — APPROVED** [review]

**Pull request merged**